security_privacy_ln.asciidoc

```diff
@@ -0,0 +1,614 @@
+
+One of the most subtle and yet powerful leakages an on-path adversary can foster is the timing analysis.
+An on-path adversary can log for every routed payment along with the amount of time it takes for a node to respond to an HTLC request.
+Before starting the attack, the attacker learns every node's latency characteristics in the Lightning Network by sending them requests.
+Naturally, this can aid in establishing the adversary's precise position in the payment path.
+Even more, as it was recently shown, an attacker can successfully determine the sender and the receiver of a payment from a set of possible senders and receivers using time-based estimators.
+
+Finally, we remark that several, yet unknown, leakages might exist that can aid deanonymizing attempts.
+For instance, even knowing the applied routing algorithm could help exclude certain nodes from being a sender and/or receiver of a payment.
+We note that different Lightning wallets apply different routing algorithms.
+Likely, many more leakages exist.
+
+== Revealing channel balances (probing)
+
+The balances of Lightning channels are supposed to be hidden for privacy and efficiency reasons.
+A Lightning node only knows the balances of its adjacent channels.
+The protocol provides no standard way to query the balance of a remote channel.
+
+However, an attacker can reveal the balance of a remote channel in a _probing attack_.
+In information security, probing refers to the technique of sending requests to a targeted system and making conclusions about its private state based on the received responses.
+
+Lightning channels are prone to probing.
+Recall that a standard Lightning payment starts with the receiver creating a random payment secret and sending its hash to the sender.
+Note for the intermediary nodes, all hashes look random.
+There is no way to tell whether a hash corresponds to a real secret or was generated randomly.
+
+The probing attack proceeds as follows.
+Say, the attacker Mallory wants to reveal Alice's balance of a public channel between Alice and Bob.
+Suppose the total capacity of that channel is 1 million satoshis.
+Alice's balance could be anything from zero to 1 million satoshis (to be precise, the estimate is a bit tighter due to channel reserve, but we don't account for it here for simplicity).
+Mallory opens a channel with Alice with 1 million satoshis and sends 500 thousand satoshis to Bob via Alice using a _random number_ as the payment hash.
+Of course, this number does not correspond to any known payment secret.
+Therefore, the payment will fail.
+The question is: how exactly will it fail?
+
+There are two scenarios.
+If Alice has more than 500 thousand satoshis on her side of the channel to Bob, she forwards the payment.
+Bob decrypts the payment onion and realizes that the payment is intended for him.
+He looks up his local store of payment secrets, searches for the preimage that corresponds to the payment hash but does not find one.
```

Comment:
Use `pre-image` instead of `preimage` for consistency as the first version was used previously.

```suggestion
He looks up his local store of payment secrets, searches for the pre-image that corresponds to the payment hash but does not find one.
```

Reply:
Changed it to "preimage" everywhere.